addons/website_slides/models/slide_slide.py

```diff
@@ -700,165 +816,231 @@ def _compute_quiz_info(self, target_partner, quiz_done=False):
-            values['binary_content'] = self._fetch_data(google_values['webContentLink'], {}, 'pdf')['values']
-            values['slide_category'] = get_slide_category(values)
-
-        return {'values': values}
+
+        if google_drive_values.get('thumbnailLink'):
+            # small trick, we remove '=s220' to get a higher definition
+            thumbnail_url = google_drive_values['thumbnailLink'].replace('=s220', '')
+            if fetch_image:
+                slide_metadata['image_1920'] = base64.b64encode(
+                    requests.get(thumbnail_url, timeout=3).content
+                )
+            else:
+                slide_metadata['image_url'] = thumbnail_url
+
+        if self.slide_category == 'document':
+            sheet_mimetypes = [
+                'application/vnd.ms-excel',
+                'application/vnd.openxmlformats-officedocument.spreadsheetml.sheet',
+                'application/vnd.oasis.opendocument.spreadsheet',
+                'application/vnd.google-apps.spreadsheet'
+            ]
+
+            doc_mimetypes = [
+                'application/msword',
+                'application/vnd.openxmlformats-officedocument.wordprocessingml.document',
+                'application/vnd.oasis.opendocument.text',
+                'application/vnd.google-apps.document'
+            ]
+
+            slides_mimetypes = [
+                'application/vnd.ms-powerpoint',
+                'application/vnd.openxmlformats-officedocument.presentationml.presentation',
+                'application/vnd.oasis.opendocument.presentation',
+                'application/vnd.google-apps.presentation'
+            ]
+
+            mime_type = google_drive_values.get('mimeType')
+            if mime_type == 'application/pdf':
+                slide_metadata['slide_type'] = 'pdf'
+                if google_drive_values.get('downloadUrl'):
+                    # attempt to download PDF content to extract a completion_time based on the number of pages
+                    try:
+                        pdf_response = requests.get(google_drive_values.get('downloadUrl'), timeout=5)
+                        completion_time = self._get_completion_time_pdf(pdf_response.content)
+                        if completion_time:
+                            slide_metadata['completion_time'] = completion_time
+                    except Exception:
+                        pass  # fail silently as this is nice to have
+            elif mime_type in sheet_mimetypes:
+                slide_metadata['slide_type'] = 'sheet'
+            elif mime_type in doc_mimetypes:
+                slide_metadata['slide_type'] = 'doc'
+            elif mime_type in slides_mimetypes:
+                slide_metadata['slide_type'] = 'slides'
+            elif mime_type and mime_type.startswith('image/'):
+                # image and videos should be input using another "slide_category" but let's be nice and
```

Comment:
Won't we update the slide_category ?

Reply:
It would be very hard to support on the website form.
A bit of a side-case really, not sure we should go into complicated handling.